minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java

```diff
@@ -102,9 +110,13 @@
    */
   private boolean mSuccess;
 
-  private MultiProcessCluster(Map<PropertyKey, String> properties, int numMasters, int numWorkers,
-      String clusterName, DeployMode mode) {
+  private MultiProcessCluster(Map<PropertyKey, String> properties,
+      Map<Integer, Map<PropertyKey, String>> masterProperties,
+      Map<Integer, Map<PropertyKey, String>> workerProperties,
+      int numMasters, int numWorkers, String clusterName, DeployMode mode) {
     mProperties = properties;
+    mMasterProperties = masterProperties;
```

Comment:
can we verify in the constructor that all the keys in masterProperties are less than numMasters, and same for workerProperties?

Reply:
Will change to
```
    public MultiProcessCluster build() {
      Preconditions.checkState(mMasterProperties.keySet()
              .stream().filter(key -> (key >= mNumMasters || key < 0)).count() == 0,
          "The master indexes in master properties should be bigger or equal to zero "
              + "and small than %s", mNumMasters);
      Preconditions.checkState(mWorkerProperties.keySet()
              .stream().filter(key ->  (key >= mNumWorkers || key < 0)).count() == 0,
          "The worker indexes in worker properties should be bigger or equal to zero "
              + "and small than %s", mNumWorkers);
      return new MultiProcessCluster(mProperties, mMasterProperties, mWorkerProperties,
          mNumMasters, mNumWorkers, mClusterName, mDeployMode);
    }
```
Thanks